### PR TITLE
V pritka/bot me message reminder nodejs

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**
ME Action which allows sending 1:1 notification with link to the selected message.

### CODE REVIEW

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+## Message reminder with messaging extension action
+
+This sample shows feature where user can schedule a task from messaging extension action and get a reminder card at scheduled time.
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:3]: This sample shows a feature where user can schedule a task from messaging extension action and get a reminder card at a scheduled time.

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+![Select message](Images/MessageAction.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:5]: Can you please crop this image and show only relevant section:
image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+![Task Details](Images/ScheduleTask.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:7]: Why the task title is not showing in a textbox like for description ??

image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+![Task reminder](Images/TaskReminder.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:9]: I am not sure why are we passing only Title here without and prefix like 
"**Task with Title:** Development call reminder"

image

Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:9]: Also there is no proper spacing after colon (:)

image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+![Team message action ](Images/MessageActionTeam.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:11]: Kindly crop this image and add only relevant section here.

image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+![Team Task Details ](Images/ScheduleTaskTeam.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:13]: Similar image at above, you are using Textbox for description and not in Title.
But here I can't see the text boxes properly.

Could you please update the UI properly.
image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+![ TeamTask reminder](Images/TaskReminderTeam.png)
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:15]: You can mark Title and subtitle in bold to differentiate and also add space after colon (:)
image

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=csharp%2Caadv2).
+
+### 2. Run your bot sample
+1) Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+2) In a terminal, navigate to `samples/bot-message-reminder/nodejs`
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/Readme.md:40]: Which terminal??
Where will the user open the code in VS or VS Code and how?

File: samples/bot-message-reminder/nodejs/appPackage/README.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# App Package
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/appPackage/README.md:1]: Is this file required??

File: samples/bot-message-reminder/nodejs/appPackage/manifest.json [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  "id": "{{Microsoft-App-Id}}",
+  "packageName": "com.microsoft.teams.messagingextensionmessagereminder",
+  "developer": {
+    "name": "Microsoft",
+    "websiteUrl": "https://www.microsoft.com",
+    "privacyUrl": "https://www.microsoft.com/privacy",
+    "termsOfUseUrl": "https://www.microsoft.com/termsofuse"
+  },
+  "name": {
+    "short": "Message reminder from ME",
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/appPackage/manifest.json:14]: Message reminder using messaging extension

File: samples/bot-message-reminder/nodejs/appPackage/manifest.json [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    "full": "Message reminder from ME"
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/appPackage/manifest.json:15]: Message reminder setup through messaging extension action command.

File: samples/bot-message-reminder/nodejs/appPackage/manifest.json [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  "description": {
+    "short": "Message reminder from ME",
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/appPackage/manifest.json:18]: Message reminder setup through messaging extension action command

File: samples/bot-message-reminder/nodejs/appPackage/manifest.json [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    "full": "This sample shows feature where user can schedule a task from messaging extension action and get a reminder card at scheduled date and time."
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/appPackage/manifest.json:19]: This sample shows a feature where user can schedule a task from messaging extension action and get a reminder card at a scheduled date and time.

File: samples/bot-message-reminder/nodejs/bots/teamsBot.js [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+var conversationReferences = {};
+var adapter;
+
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+        this.baseUrl = process.env.BaseUrl;
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (let member = 0; member < membersAdded.length; member++) {
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/bots/teamsBot.js:17]: One line space before it.

File: samples/bot-message-reminder/nodejs/models/taskModuleResponseFactory.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+class TaskModuleResponseFactory {
+    static createResponse(taskModuleInfoOrString) {
+        if (typeof taskModuleInfoOrString === 'string') {
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/models/taskModuleResponseFactory.js:6]: One line space before it.

File: samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+            .container{
+                margin-top: 1rem;
+                margin-left: 1.5rem;
+            }
+        </style>
+    
+        <script>
+            var title;
+            var description;
+            microsoftTeams.initialize(()=>{
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs:21]: One line space before it.

File: samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+                const params = window.location.search;
+                title = params.substring(params.search("=")+1,params.search("&"));
+                description=params.substring(params.search("&")+13);
+                $("#taskTitle").text(title.replace(/%20/g," "));
+                if(description == "") {
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs:26]: One line space before it.

File: samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+                        });
+                    }
+                    else {
+                        $(this).css({
+                            "border": "",
+                            "background": ""
+                        });
+                    }
+                });
+                if (isValid == false) {
```
Comment (v-chetsh) [inline samples/bot-message-reminder/nodejs/views/ScheduleTask.ejs:51]: One line space before it.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed and provided my feedback.

**Review by v-chetsh — APPROVED** [review]
Changes looks fine approving the PR.

File: README.md [hunk 19]
```diff
@@ -237,6 +238,7 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 [appcheckinlocation#csharp]:samples/app-checkin-location/csharp
 [checkinlocation#nodejs]:samples/app-checkin-location/nodejs
 [productinspection#nodejs]:samples/tab-product-inspection/nodejs
+[memessagereminder#nodejs]:samples/bot-message-reminder/nodejs
```
Comment (Wajeed-msft) [inline README.md:241]: Please renmae the folde to msgext-message-reminder. Please do the same for node project.

Comment (Wajeed-msft) [inline README.md:241]: Please renmae the folde to msgext-message-reminder.

File: samples/bot-message-reminder/nodejs/Readme.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
+## Message reminder with messaging extension action
```
Comment (Wajeed-msft) [inline samples/bot-message-reminder/nodejs/Readme.md:1]: 1. Add SEO metadata.
2. Keep header with single #
3. Keep only first three images.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**